src/dbnode/storage/shard.go

```diff
@@ -1890,10 +1898,13 @@ func (s *dbShard) WarmFlush(
 		NamespaceMetadata: s.namespace,
 		Shard:             s.ID(),
 		BlockStart:        blockStart,
+		// Volume index is always 0 for warm flushes because a warm flush must
+		// happen first before cold flushes happen.
+		VolumeIndex: 0,
 		// We explicitly set delete if exists to false here as we track which
-		// filesets exists at bootstrap time so we should never encounter a time
-		// when we attempt to flush and a fileset already exists unless there is
-		// racing competing processes.
+		// filesets exist at bootstrap time so we should never encounter a time
+		// when we attempt to flush and a fileset that already exists unless
+		// there are racing competing processes.
```

Comment:
"there is a bug in the code" or something

Reply:
Haha sure.